hcl/eval/eval.go

```diff
@@ -107,39 +107,40 @@ func (c *Context) PartialEval(expr hclsyntax.Expression) (hclwrite.Tokens, error
 	return engine.Eval()
 }
 
-func toWriteTokens(in hclsyntax.Tokens) hclwrite.Tokens {
-	tokens := make([]*hclwrite.Token, len(in))
-	for i, st := range in {
-		tokens[i] = &hclwrite.Token{
-			Type:  st.Type,
-			Bytes: st.Bytes,
-		}
-	}
-	return tokens
-}
-
-func fromMapToObject(m map[string]cty.Value) (cty.Value, error) {
+// FromMapToObject converts a map of cty.Value to an object.
+func FromMapToObject(values map[string]cty.Value) (cty.Value, error) {
 	logger := log.With().
 		Str("action", "fromMapToObject()").
 		Logger()
 
-	logger.Trace().
-		Msg("Range over map.")
+	logger.Trace().Msg("Range over map")
+
 	ctyTypes := map[string]cty.Type{}
-	for key, value := range m {
+	for key, value := range values {
 		ctyTypes[key] = value.Type()
 	}
 
-	logger.Trace().
-		Msg("Convert type and value to object.")
+	logger.Trace().Msg("Convert type and value to object")
+
 	ctyObject := cty.Object(ctyTypes)
-	ctyVal, err := gocty.ToCtyValue(m, ctyObject)
+	ctyVal, err := gocty.ToCtyValue(values, ctyObject)
```

Comment:
I'm torn if we shouldn't just panic here. The conversion always uses terramate evaluated things (or built by ourselves some other way), so feels like a bug/critical thing going on that could be panicked (would simplify a lot the error handling around this). WDYT @i4ki ?

Reply:
I think makes sense, as this a controlled namespace.

Reply:
Will improve this on a follow up PR.